src/screens/PostScreen.tsx

```diff
@@ -1,43 +1,117 @@
-import { Text, View } from 'react-native';
+import React, { useEffect, useState } from 'react';
+import { Image, Text, View } from 'react-native';
+import supabase from 'supabase/client';
 import HeartIcon from '../../assets/heart-icon.svg';
 import ShareIcon from '../../assets/messenger-icon.svg';
 import ProfilePlaceholder from '../../assets/profile-placeholder-icon.svg';
 import { styles } from './styles';
 
 export default function PostScreen() {
+  interface Post {
+    id: number;
+    created_at: string;
+    description: string;
+    username: string;
+    image_url: string;
+    like_count: number;
+  }
+
+  const getPostData = async (): Promise<Post[]> => {
+    const { data, error } = await supabase
+      .from('posts')
+      .select('*')
+      .eq('username', 'rbeggs');
+
+    if (error) {
+      throw new Error(`Error fetching post: ${error.message}`);
+    }
+    return data as Post[];
+  };
+
+  const [postData, setPostData] = useState<Post[]>([]);
+
+  useEffect(() => {
+    fetchPostData();
+  }, []);
+
+  const fetchPostData = async () => {
+    try {
+      const data = await getPostData();
+      setPostData(data);
+    } catch (error) {
+      console.error('Error fetching data:', error);
+    }
+  };
```

Comment:
I don't know if you necessarily need a `fetchPostData` and `getPostData` function in the same file. Usually you'd store the `getPostData` function in a `/supabase/post.ts` queries file.